services/consensusalgo/leanhelixconsensus/service.go

```diff
@@ -183,7 +183,8 @@ func (s *Service) validateBlockExecutionIfYoung(ctx context.Context, blockPair *
 			CurrentBlockHeight: blockPair.TransactionsBlock.Header.BlockHeight(),
 			PrevBlockHash:      blockPair.TransactionsBlock.Header.PrevBlockHashPtr(),
 			TransactionsBlock:  blockPair.TransactionsBlock,
-			PrevBlockTimestamp: prevBlockPair.TransactionsBlock.Header.Timestamp()})
+			PrevBlockTimestamp: prevBlockPair.TransactionsBlock.Header.Timestamp(),
+			BlockProposerAddress: blockPair.TransactionsBlock.Header.BlockProposerAddress()}) // TODO NOAM maybe need to ask lean helix lib for proposer
```

Comment:
@gadcl - (1) any reason block proposer would be different between TransactionBlock and ResultsBlock? If block proposer is always the same then this code is good and can remove the TODO comment. (2) If it's a duplicate of the same field in Transactions Block, can we just remove this field from Results Block?

Reply:
consistent with the current `block pair` design, as to support future split of execution and ordering processes.